CRM/Event/Form/Registration.php

```diff
@@ -560,10 +560,10 @@ public function assignToTemplate() {
 
     $this->assign('is_email_confirm', $this->_values['event']['is_email_confirm'] ?? NULL);
     // assign pay later stuff
-    $params['is_pay_later'] ??= FALSE;
-    $this->assign('is_pay_later', $params['is_pay_later']);
-    $this->assign('pay_later_text', $params['is_pay_later'] ? $this->getPayLaterLabel() : FALSE);
-    $this->assign('pay_later_receipt', $params['is_pay_later'] ? $this->_values['event']['pay_later_receipt'] : NULL);
+    $isPayLater = ($this->_values['event']['is_pay_later'] ?? FALSE);
```

Comment:
$params actually comes from the first event participant, we should be looking at the actual event data.

Reply:
I've been using `$this->getEventValue('is_pay_later')` rather than `_values` - the former is a direct apiv4 lookup with some caching whereas the latter seems a bit magic to me

Reply:
(also `getEventValue()` is supported for extensions to use so good to model it)

Reply:
It's pretty horrible the way event/participant params are passed around those forms. The problem is that *both* participant params and event (_values) are modified during form submission. The database value of is_pay_later is 1 for the event because pay later is enabled. But by the time we get to this point is_pay_later has been set to FALSE because we've submitted the form with a paymentprocessor. If you use the event value from the DB/API4 then you get contributions etc. with is_pay_later set to true when they are not..
I'd love to find another way of getting the form submission is_pay_later value but I couldn't find anything that didn't involve a massive refactoring of the params being passed around.

Reply:
oh that's  dark - but does that mean `is_pay_later` === `!$this->getSubmittedValue('payment_processor_id')` ?

Reply:
Oh, that's an idea.. getSubmittedValue() wasn't working properly but I fixed that as part of this PR so may now be a better option!